### ISSUE TITLE
Created Jar is failing to execute via java -jar

### ISSUE THREAD

**Issue by gamodg** [camunda-community-hub/camunda-platform-7-rest-client-java#25]
Used the project at https://github.com/berndruecker/camunda-7-remote-spring-boot-example
- Did mvn package to create jar
- Ran `java -jar <APP>.jar` got the error 
`no main manifest attribute, in camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar`

Updated pom.xml to have 
```
<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<version>2.1.5.RELEASE</version>
				<executions>
					<execution>
						<goals>
							<goal>repackage</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>
```

but now I am getting below error. Please help !!
```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.camunda.community.rest.client.springboot.CamundaProcessAutodeployment': Invocation of init method failed; nested exception is java.io.FileNotFoundException: URL [jar:file:/C:/Users/test/Downloads/camunda-7-remote-spring-boot-example-main/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.bpmn] cannot be resolved to absolute file path because it does not reside in the file system: jar:file:/C:/Users/test/Downloads/camunda-7-remote-spring-boot-example-main/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.bpmn
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:160) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:440) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1796) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[spring-context-5.3.19.jar!/:5.3.19]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.19.jar!/:5.3.19]
        at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:740) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:415) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1312) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301) ~[spring-boot-2.6.7.jar!/:2.6.7]
        at io.berndruecker.demo.springboot.SpringBootApp.main(SpringBootApp.java:15) ~[classes!/:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) ~[camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) ~[camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) ~[camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:51) ~[camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar:na]
Caused by: java.io.FileNotFoundException: URL [jar:file:/C:/Users/test/Downloads/camunda-7-remote-spring-boot-example-main/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.bpmn] cannot be resolved to absolute file path because it does not reside in the file system: jar:file:/C:/Users/test/Downloads/camunda-7-remote-spring-boot-example-main/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.bpmn
        at org.springframework.util.ResourceUtils.getFile(ResourceUtils.java:217) ~[spring-core-5.3.19.jar!/:5.3.19]
        at org.springframework.core.io.AbstractFileResolvingResource.getFile(AbstractFileResolvingResource.java:162) ~[spring-core-5.3.19.jar!/:5.3.19]
        at org.springframework.core.io.UrlResource.getFile(UrlResource.java:240) ~[spring-core-5.3.19.jar!/:5.3.19]
        at org.camunda.community.rest.client.springboot.CamundaProcessAutodeployment.deployCamundaResources(CamundaProcessAutodeployment.java:54) ~[camunda-engine-rest-client-openapi-springboot-7.17.0.jar!/:7.17.0]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:333) ~[spring-beans-5.3.19.jar!/:5.3.19]
        at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:157) ~[spring-beans-5.3.19.jar!/:5.3.19]
        ... 26 common frames omitted
```

**Comment by vijamesDD** [comment]
Yes I am also having this exact issue as well! I've not been able to run this project as a jar. I've debugged into the inner workings of the resource loader and it appears that during autodeployment of BPMN files the resource loader recognises the bpmn file as a jar rather than a file. (note the "jar" at the beginning of the URL in @gamodg's stack trace above). 

When the app is ran using spring boots runner (ie `mvn clean spring-boot:run`) this error is not thrown, as it seems the resource loader correctly recognises the bpmn files as Files rather than jars. However unfortunately I need to run this app as a packaged Jar and using spring-boots runner is not an option. 

@berndruecker any ideas please?

**Comment by githanium** [comment]
same issue here

**Comment by gamodg** [comment]
As a workaround followed the article to use layers for ensuring that files are not packaged in a jar
https://spring.io/blog/2020/01/27/creating-docker-images-with-spring-boot-2-3-0-m1

It is working fine for me 
`FROM adoptopenjdk/openjdk11-openj9:jdk-11.0.14_9_openj9-0.30.0-alpine-slim as builder
WORKDIR /app
ARG JAR_FILE=/target/*.jar
COPY ${JAR_FILE} app.jar
RUN java -Djarmode=layertools -jar app.jar extract

FROM adoptopenjdk/openjdk11-openj9:jre-11.0.14_9_openj9-0.30.0-alpine
WORKDIR /app
COPY --from=builder app/dependencies/ ./
COPY --from=builder app/snapshot-dependencies/ ./
COPY --from=builder app/spring-boot-loader/ ./
COPY --from=builder app/application/ ./
ENTRYPOINT ["java", "org.springframework.boot.loader.JarLauncher"]
`

**Comment by berndruecker** [comment]
I wonder why you want to `java -jar` this artifact? What exactly do you want to achieve?

This library is meant to be used within your own application as library/SDK (like e.g. https://github.com/camunda-community-hub/camunda-platform-7-rest-client-java#spring-boot-starter). Then you need to start your own app - and not this library - or do I get this wrong?

**Comment by vijamesDD** [comment]
Thanks @berndruecker  for your response! Perhaps I am misunderstanding your message, but we are not packaging and trying to run this artefact individually. 

We are pulling this library into our own application by setting the dependency in our pom file as per the instructions in the [readme of this repo](https://github.com/camunda-community-hub/camunda-platform-7-rest-client-java#spring-boot-openapi--external-task-bundle), and then attempting to package and run our application as a JAR file. 

The error (see stack trace in issue description) can be reproduced by taking the [example project from the readme](https://github.com/berndruecker/camunda-7-remote-spring-boot-example) which makes use of this repository as a dependency, adding the spring repackage plugin to the pom (as per description of issue) to enable packaging into a jar, then running:
`java -jar <PATH TO JAR>/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar`

**Comment by berndruecker** [comment]
Sorry - my bad - understood now. I added the proper repackaging rule for Spring Boot in the example: https://github.com/berndruecker/camunda-7-remote-spring-boot-example/blob/main/pom.xml#L26

Note that you should update to 7.17.2 (https://github.com/berndruecker/camunda-7-remote-spring-boot-example/blob/main/pom.xml#L16) because that fixes some resource deployment problems in this setup.

**Comment by berndruecker** [comment]
@vijamesDD Thanks for reporting! Does it help if I say this line can't be null ;-) 

Jokes aside - This is a bit puzzling to me. I can easily fix it (https://github.com/camunda-community-hub/camunda-platform-7-rest-client-java/commit/d3ddc3d30eb2801dcdcdf251f9fd85f1aa27a3e5) - but I still don't understand it - so I don't know how to build a test case for it. 

How do you run the `camunda-platform-remote-spring-boot-example` - via IDE or `java -jar`?
Which operating system are you running?

Unfortunately, the OpenAPI client needs a File (instead of an InputStream) - so there might be differences on this level. 

Can you test against the latest main branch (so a 7.13.0-SNAPSHOT)?
Thx a lot!

**Comment by berndruecker** [comment]
Released 7.13.3 with this fix - can you double check if that works for you?

**Comment by vijamesDD** [comment]
Hi @berndruecker , thanks for getting this change in! I've tested version 7.12.3 as that appears to be the version thats most recently deployed (incidentally did you mean for it to be 7.**12**.3 ? the version before that was 7.17.2 I believe)
With 7.12.3 of this lib in the example app I can now start it up successfully using either a packaged jar or spring runner! 👏 🎉 ie either:
1. `mvn clean spring-boot:run`
2. `mvn clean spring-boot:repackage && java -jar <PATH TO JAR>/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar`
Thank you!!

I noticed though that the bpmn files in the resource folder do not appear to be auto deployed to the engine service on startup, does that occur for you as well?

**Comment by berndruecker** [comment]
The version was a typo from my end - sorry for that. Let me have a look into the deployment tomorrow - I thought it is picking it up properly - but I will double check and probably also add some logging to better diagnose problems

**Comment by berndruecker** [comment]
I can't reproduce your problem - when I run 
`java -jar .\target\camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar`

Then I can also see the deployment happing with the just built-in logging:

`
2022-07-14 11:10:58.530  INFO 21288 --- [           main] o.c.c.r.c.s.CamundaProcessAutodeployment : Found resources for deployment: [URL [jar:file:/C:/DEV/Camunda/camunda-7-remote-spring-boot-example/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.b
pmn], URL [jar:file:/C:/DEV/Camunda/camunda-7-remote-spring-boot-example/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground.bpmn], URL [jar:file:C:\DEV\Camunda\camunda-7-remote-spring-boot-example\.\target\camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHO
T.jar!/BOOT-INF/classes/playground-sempahore.bpmn], URL [jar:file:C:\DEV\Camunda\camunda-7-remote-spring-boot-example\.\target\camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes/playground.bpmn]]
2022-07-14 11:10:58.541  INFO 21288 --- [           main] o.c.c.r.c.s.CamundaProcessAutodeployment :   - Now deploying: URL [jar:file:/C:/DEV/Camunda/camunda-7-remote-spring-boot-example/target/camunda-platform-remote-spring-boot-example-0.0.1-SNAPSHOT.jar!/BOOT-INF/classes!/playground-sempahore.bpmn]
`

This is a bit different to when running the application via IDE - which is in-line with expectations:

`
2022-07-14 11:14:24.427  INFO 11060 --- [           main] o.c.c.r.c.s.CamundaProcessAutodeployment : Found resources for deployment: [file [C:\DEV\Camunda\camunda-7-remote-spring-boot-example\target\classes\playground-sempahore.bpmn], file [C:\DEV\Camunda\camunda-7-remote-spring-boot-example\target\classes\playground.bpmn]]
2022-07-14 11:14:24.440  INFO 11060 --- [           main] o.c.c.r.c.s.CamundaProcessAutodeployment :   - Now deploying: file [C:\DEV\Camunda\camunda-7-remote-spring-boot-example\target\classes\playground-sempahore.bpmn]
`

Could you retry with the now correctly built 7.17.3?

**Comment by vijamesDD** [comment]
Hi @berndruecker , sure thing! I've tested with 7.17.3 in the example application. I am also seeing those log lines, but I don't think the auto-deployment is working as intended 😢  I'm finding that the remote engine is not recognising the files as processes and is therefore not creating process definitions for them. Could this be because the [`.bpmn` files are being mapped to `.tmp`](https://github.com/camunda-community-hub/camunda-platform-7-rest-client-java/blob/025190debafafc5a098b0bef7f6b92cc3e39bef9/camunda-engine-rest-client-openapi-springboot/src/main/java/org/camunda/community/rest/client/springboot/CamundaProcessAutodeployment.java#L71) files before being sent in the create deployment request? 

Screnshots below of the camunda cockpit for my locally running engine which appears to not have the expected processes defined... 
<img width="1693" alt="Screenshot 2022-07-14 at 17 31 49" src="https://user-images.githubusercontent.com/26329741/179032134-026682a8-1bbc-4566-be53-88d4f913b9ba.png">
<img width="1696" alt="Screenshot 2022-07-14 at 17 32 03" src="https://user-images.githubusercontent.com/26329741/179032276-6ae4d360-0133-4e94-890c-baf3ac8bc84e.png">
<img width="1691" alt="Screenshot 2022-07-14 at 17 32 09" src="https://user-images.githubusercontent.com/26329741/179032440-efaf77c0-1170-4264-b844-8ad8a2fecb99.png">

**Comment by berndruecker** [comment]
Thanks for the additional context. Got it - and you are right. It is because of the file ending - my bad (probably not a good idea to do these things as a side hobby whenever there is a little time slot). Nevertheless fixed now in 7.17.4 and tried successfully end-to-end on my end.
